V2 task service refactoring

anleo opened the pull request.

anleo committed:
- fixed TasksBoardFilter pipe
- fixed relations between taskService, taskEdit and taskItem components
- cleanup
- removed task modal component

anleo committed:
Merge branch 'v2' into v2-task-service-refactoring

anleo committed:
- fixed TasksBoardFilter pipe
- fixed relations between taskService, taskEdit and taskItem components
- cleanup
- removed task modal component

13bit commented on public-v2/src/app/tracker/services/task.service.ts:
```diff
@@ -6,51 +6,122 @@ import {Task} from '../models/task';
+    this.editTaskUpdated$.subscribe((taskWithStatus: TaskWithStatus) => this.updater(taskWithStatus));
+  }
+
+  updater(taskWithStatus: TaskWithStatus):void|boolean {
```
может переименовать?

actionProvider

13bit commented on public-v2/src/app/tracker/services/task.service.ts:
```diff
@@ -6,51 +6,122 @@ import {Task} from '../models/task';
 import {FileResourse} from "../resources/file.resource";
+import {Router} from "@angular/router";
+
+class TaskWithStatus {
```
Вынести в отдельный класс

13bit commented on public-v2/src/app/tracker/services/task.service.ts:
```diff
@@ -6,51 +6,122 @@ import {Task} from '../models/task';
+  }
+
+  onMove(task: Task):void {
+    if (task) {
```
может уменьшить вложенность и пойти от обратного ?

13bit commented on public-v2/src/app/tracker/tasks/components/task-move/task-move.component.ts:
```diff
@@ -25,8 +25,7 @@ export class TaskMoveComponent implements OnInit {
       .subscribe(task => {
-        this.taskService.setMovedTask(task);
-        this.onMoved.emit(task);
+        this.onMove.emit(task);
```
сделай одной строкой

13bit commented on public-v2/src/app/tracker/tasks/task-item/task-item.component.ts:
```diff
@@ -32,36 +35,29 @@ export class TaskItemComponent implements OnInit {
-          .subscribe(task => this.onTaskMoved(task));
-
-        this.init(task);
+        this.initTaskData(task);
```
в одну строку

13bit commented on public-v2/src/app/tracker/tasks/task-item/task-item.component.ts:
```diff
@@ -32,36 +35,29 @@ export class TaskItemComponent implements OnInit {
-        this.onRemove(task);
-      }
+    this.loadTasks(taskId).subscribe(tasks => {
+      this.taskService.tasks$.next(tasks);
```
в одну строку

13bit commented on public-v2/src/app/tracker/tasks/tasks-list/tasks-list.component.ts:
```diff
@@ -10,7 +9,7 @@ import {TaskService} from "../../services/task.service";
-  constructor(private taskService:TaskService) { }
+  constructor() { }
 
   ngOnInit() {
```
убрать NgOnInit который не используется

13bit commented on public-v2/src/app/tracker/tasks/tasks.component.ts:
```diff
@@ -11,12 +10,10 @@ export class TasksComponent implements OnInit {
 
-  constructor(private taskService:TaskService) { }
+  constructor() {
+  }
```
убрать пустой конструктор и onInit

anleo committed:
- fixed TasksBoardFilter pipe
- fixed relations between taskService, taskEdit and taskItem components
- cleanup
- removed task modal component
